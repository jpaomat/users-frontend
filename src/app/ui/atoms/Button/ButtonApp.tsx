import PropTypes from 'prop-types';
import './ButtonApp.scss';
import { IButtonApp } from '../../interfaces';

export const ButtonApp = ({ children, size, variant, isDisabled, onClick }: IButtonApp) => {
	return (
		<button
			className={`btn btn-${variant} btn-${size}`}
			disabled={isDisabled}
			onClick={onClick}
		>
			{children}
		</button>
	);
};

ButtonApp.defaultProps = {
	isDisabled: false,
	size: 'sm',
	variant: 'primary',
};

ButtonApp.propTypes = {
	children: PropTypes.element.isRequired,
	isDisabled: PropTypes.bool,
	onClick: PropTypes.func.isRequired,
	size: PropTypes.string,
	variant: PropTypes.string,
};
