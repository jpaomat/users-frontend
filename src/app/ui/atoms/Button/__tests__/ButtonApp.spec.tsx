import { fireEvent, render, screen } from '@testing-library/react';
import { ButtonApp } from '../ButtonApp';
import { defaultMock } from './mocks';

describe('<ButtonApp /> component', () => {
	const buttonText = 'Continuar';

	beforeEach(()=> jest.clearAllMocks);
	
	test('should render component', () => {
		const { container } = render(
			<ButtonApp {...defaultMock}>
				<span>{buttonText}</span>
			</ButtonApp>
		);
		expect(container).toMatchSnapshot();
	});

	test('should show the text Continuar', () => {
		render(
			<ButtonApp {...defaultMock}>
				<span>{buttonText}</span>
			</ButtonApp>
		);
		fireEvent.click(screen.getByText(buttonText));
		expect(screen.getByText(buttonText)).toBeTruthy();
	});

	test('should execute the onClick event that is sent in the props', () => {
		let testEvent = '';
		const onClickTest = () => {
			testEvent = 'Hi Button';
		};
		render(
			<ButtonApp {...defaultMock} onClick={onClickTest}>
				<span>{buttonText}</span>
			</ButtonApp>
		);
		fireEvent.click(screen.getByText(buttonText));
        expect(testEvent).toEqual('Hi Button');
	});
});
