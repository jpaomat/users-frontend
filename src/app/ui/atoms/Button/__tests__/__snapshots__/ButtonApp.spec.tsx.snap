// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonApp /> component should render component 1`] = `
<div>
  <button
    class="btn btn-danger btn-sm"
  >
    <span>
      Continuar
    </span>
  </button>
</div>
`;
